neotest_elixir/neotest_formatter.exs

```diff
@@ -128,6 +129,41 @@ defmodule NeotestElixirFormatter do
 
   defp make_output(%ExUnit.Test{}, _config), do: nil
 
+  defp make_errors(%ExUnit.Test{state: {:failed, failures}} = test) do
+    Enum.map(failures, fn failure ->
+      %{message: make_error_message(failure), line: make_error_line(failure, test)}
+    end)
+  end
+
+  defp make_errors(%ExUnit.Test{}), do: []
+
+  defp make_error_message(failure) do
+    case failure do
+      {:error, exception, _stack} when is_exception(exception) ->
+        exception |> Exception.message() |> String.trim()
```

Comment:
This seems to include `message`, `expr`, `left`, `right` in the case of `ExUnit.AssertionError`. 
For assertion failures on say large struct / map comparisons this is not useful. (and everything is shown in red)
Given that we anyways show output window for failures(with appropriate diff colouring), it might be better to just stick with `message` and maybe `expression` (ignoring the `left` and `right`). But I guess this can be done in a separate PR too
The default formatter [code](https://github.com/elixir-lang/elixir/blob/df935619d90ab08ee84fe8f16bf7cbf8e8c6fc87/lib/ex_unit/lib/ex_unit/formatter.ex#L224-L227) for reference

Reply:
Yes, I think long messages in red don't help much. I was thinking we could split the message by `\n\n`, and take the first part, so we only get the first block. Which usually tells enough information to know what failed.